app/test/ui/views/signup/sign_up_screen_test.dart

```diff
@@ -0,0 +1,208 @@
+          find.widgetWithText(RaisedButton, "Register");
+      await tester.ensureVisible(registerButtonFinder);
+      await tester.tap(registerButtonFinder);
+      // Non-standard animation time due to decoration
+      await tester.pump(Duration(milliseconds: 1));
+
+      final validatorTextFinder = find.text("* Required");
+      expect(validatorTextFinder, findsNWidgets(4));
+
+      verifyNever(mockSignUpViewModel.signUp(
+          userEmailAddress: anyNamed("userEmailAddress"),
+          password: anyNamed("password"),
+          firstName: anyNamed("firstName"),
+          lastName: anyNamed("lastName"),
+          userPhoneNumber: anyNamed("userPhoneNumber")));
+    });
+
+    testWidgets('email field validator appears on bad email',
+        (WidgetTester tester) async {
+      await pumpSignUpScreen(tester);
+
+      final emailFinder = find.byKey(ObjectKey("Email"));
+      await tester.enterText(emailFinder, "email");
+
+      await tester.pump(Duration(milliseconds: 1));
+
+      final registerButtonFinder =
+          find.widgetWithText(RaisedButton, "Register");
+      await tester.ensureVisible(registerButtonFinder);
+      await tester.tap(registerButtonFinder);
+      // Non-standard animation time due to decoration
+      await tester.pump(Duration(milliseconds: 1));
+
+      final validatorTextFinder = find.text("* Please enter a valid Email");
+      expect(validatorTextFinder, findsNWidgets(1));
+
+      verifyNever(mockSignUpViewModel.signUp(
+          userEmailAddress: anyNamed("userEmailAddress"),
+          password: anyNamed("password"),
+          firstName: anyNamed("firstName"),
+          lastName: anyNamed("lastName"),
+          userPhoneNumber: anyNamed("userPhoneNumber")));
+    });
+    testWidgets(
+        'password field validator appears on bad password - less than 6 characters',
+        (WidgetTester tester) async {
+      await pumpSignUpScreen(tester);
+
+      final emailFinder = find.byKey(ObjectKey("Password"));
+      await tester.enterText(emailFinder, "1234");
+
+      await tester.pump(Duration(milliseconds: 1));
+
+      final registerButtonFinder =
+          find.widgetWithText(RaisedButton, "Register");
+      await tester.ensureVisible(registerButtonFinder);
+      await tester.tap(registerButtonFinder);
+      // Non-standard animation time due to decoration
+      await tester.pump(Duration(milliseconds: 1));
+
+      final validatorTextFinder =
+          find.text("Password should be at least 6 characters");
+      expect(validatorTextFinder, findsNWidgets(1));
+
+      verifyNever(mockSignUpViewModel.signUp(
+          userEmailAddress: anyNamed("userEmailAddress"),
+          password: anyNamed("password"),
+          firstName: anyNamed("firstName"),
+          lastName: anyNamed("lastName"),
+          userPhoneNumber: anyNamed("userPhoneNumber")));
+    });
+
+    testWidgets(
+        'password field validator appears on bad password - greater that 10 characters',
```

Comment:
@mansiPaTeLL greater than 10 or 15?

Reply:
@holtzmak should not greater than 10. I changed it.